pkg/chunk/cache/background.go

```diff
@@ -80,21 +80,33 @@ func (c *backgroundCache) Stop() {
 	c.Cache.Stop()
 }
 
+const keysPerBatch = 100
+
 // Store writes keys for the cache in the background.
 func (c *backgroundCache) Store(ctx context.Context, keys []string, bufs [][]byte) {
-	bgWrite := backgroundWrite{
-		keys: keys,
-		bufs: bufs,
-	}
-	select {
-	case c.bgWrites <- bgWrite:
-		c.queueLength.Add(float64(len(keys)))
-	default:
-		c.droppedWriteBack.Add(float64(len(keys)))
-		sp := opentracing.SpanFromContext(ctx)
-		if sp != nil {
-			sp.LogFields(otlog.Int("dropped", len(keys)))
+	for len(keys) > 0 {
+		num := keysPerBatch
+		if num > len(keys) {
+			num = len(keys)
+		}
+
+		bgWrite := backgroundWrite{
+			keys: keys[:num],
+			bufs: bufs[:num],
+		}
+		select {
+		case c.bgWrites <- bgWrite:
+			c.queueLength.Add(float64(len(keys)))
```

Comment:
```suggestion
			c.queueLength.Add(float64(len(num)))
```

I think you should utilize `num` here. If I'm not mistaken `keys` would include all of the unwritten keys.

Reply:
Yeah it's a bug, thanks.